src/realm/util/file_mapper.cpp

```diff
@@ -253,8 +263,8 @@ void reclaim_pages()
             file_reclaim_index = 0;
         while (work_limit > 0) {
             SharedFileInfo& info = *mappings_by_file[file_reclaim_index].info;
-            auto done_for_now = reclaim_pages_for_file(info, work_limit);
-            if (done_for_now) {
+            auto should_move_to_next_file = reclaim_pages_for_file(info, work_limit);
```

Comment:
Do we need `should_move_to_next_file`? Is it not so that if the work_limit is not 0, then we should in all cases move on to the next file.